addons/calendar/tests/test_calendar.py

```diff
@@ -381,6 +381,19 @@ def _test_emails_has_attachment(self, partners):
         # no more email should be sent
         _test_one_mail_per_attendee(self, partners)
 
+        # Check for attachment when public user creates an event ( simulates scheduling appointment from an incognito window)
+        self.new_partner = self.env['res.partner'].create({
+            'name': 'partner_created_on_the_spot_by_the_appointment_form',
+            'email': 'partner_created_on_the_spot_by_the_appointment_form@gmail.com',
+        })
+        self.CalendarEvent.with_user(self.env.ref('base.public_user')).sudo().create({
+            'name': "publicUserEvent",
+            'partner_ids': [(6, False, self.new_partner.ids)],
+            'start': "2023-10-06 12:00:00",
+            'stop': "2023-10-06 13:00:00",
+        })
+        _test_emails_has_attachment(self, [self.new_partner])
```

Comment:
Would be nice to test with multiple attachments on the template as that's the trickier case